sdk/communication/Shared/src/CommunicationIdentifierSerializer.cs

```diff
@@ -10,25 +10,28 @@ internal class CommunicationIdentifierSerializer
     {
         public static CommunicationIdentifier Deserialize(CommunicationIdentifierModel identifier)
         {
-            var id = identifier.Id;
-            var kind = identifier.Kind;
+            string rawId = AssertNotNull(identifier.RawId, nameof(identifier.RawId), nameof(CommunicationIdentifierModel));
 
-            if (kind == CommunicationIdentifierKind.CommunicationUser)
-                return new CommunicationUserIdentifier(AssertNotNull(id, nameof(identifier.Id), kind));
-            if (kind == CommunicationIdentifierKind.CallingApplication)
-                return new CallingApplicationIdentifier(AssertNotNull(id, nameof(identifier.Id), kind));
-            if (kind == CommunicationIdentifierKind.PhoneNumber)
-                return new PhoneNumberIdentifier(AssertNotNull(identifier.PhoneNumber, nameof(identifier.PhoneNumber), kind), AssertNotNull(id, nameof(identifier.Id), kind));
-            if (kind == CommunicationIdentifierKind.MicrosoftTeamsUser)
+            if (identifier.CommunicationUser is CommunicationUserIdentifierModel user)
+                return new CommunicationUserIdentifier(AssertNotNull(user.Id, nameof(user.Id), nameof(CommunicationUserIdentifierModel)));
+
+            if (identifier.PhoneNumber is PhoneNumberIdentifierModel phoneNumber)
+            {
+                return new PhoneNumberIdentifier(
+                    AssertNotNull(phoneNumber.Value, nameof(phoneNumber.Value), nameof(PhoneNumberIdentifierModel)),
+                    AssertNotNull(identifier.RawId, nameof(identifier.RawId), nameof(PhoneNumberIdentifierModel)));
+            }
+
+            if (identifier.MicrosoftTeamsUser is MicrosoftTeamsUserIdentifierModel teamsUser)
```

Comment:
We could check that only one property is set but I'm fine with this "first one wins" strategy.

Reply:
I added verification for that